http/src/v0/refs/path.rs

```diff
@@ -0,0 +1,390 @@
+//! Ipfs path handling following https://github.com/ipfs/go-path/. This rests under ipfs-http for
+//! now until it's ready to be moved to `rust-ipfs`. There is a competing implementation under
+//! `libipld` which might do almost the same things, but with different dependencies. This should
```

Comment:
What are the dependency differences between the two? We should try to converge the two if possible so we don't have competing / duplicate implementations.

Reply:
This is the fourth. There are

 * `ipfs::IpfsPath` with no tests, might be more complete parsing wise, doesn't have anything to walk ipld
    * `ipfs::dag::resolve` walks ipld with ipfs::IpfsPath`
 * `libipld::path::Path` -- just a wrapper for Vec<String>
 * `libipld::dag::DagPath` does walks on Ipld, not on dag-pb, depends on IpldStore

Probably makes sense to move this functionality over to `ipfs::IpfsPath` during or after `dag.get`.

Not sure how relevant this would be in `libipld`. I think the absence of the dependency to any Store trait makes this implementation at least a good conversation piece.